### ISSUE TITLE
Station with no service in Shanghai Metro

### ISSUE THREAD

**Issue by thekingofcity** [wongchito/RailMapGenerator#217]
> Also the currently missing station *Longju Road* could leverage the station `service` field by selecting no other services.
> 
> ![image](https://user-images.githubusercontent.com/3353040/147852070-9c1731da-9ddc-4186-8c11-ad833a1c8ed7.png)
> 
> However, besides `railmap` and `indoor` canvas which can simply hide the station element, next and past info in the `runin` canvas needs some more tuning as stations can be missing all the path to the terminal and there can be a situation where all the following stations of this current station are temporarily closed.

_Originally posted by @thekingofcity in https://github.com/wongchito/RailMapGenerator/issues/213#issuecomment-1003562573_

**Comment by thekingofcity** [comment]
This feature requires two parts.

* Hide station in `railmap` and `indoor` canvas -> pretty simple and finished.
* Show next/past first stations with at least one service -> involves recursive search if consecutive stations have no service.

The second part also need to be overhauled after V5 is upgraded since `linestart` and `lineend` might be removed, so I'll update it after V5.

**Comment by XHG78999** [comment]
I think we can add the second-name like Guangzhou Metro to provide this. Or we can also use an "hidden station" (Like Yanyu Road in Line 11,it's really but it's hidden in all maps actually.In POV of Line 11 you can see it between Pusan Road and Yuqiao.)

**Comment by thekingofcity** [comment]
> I think we can add the second-name like Guangzhou Metro to provide this.

No, I'll not use a `notesGZMTR` thing as shmetro does not use second-name in future station.

> Or we can also use an "hidden station" (Like Yanyu Road in Line 11,it's really but it's hidden in all maps actually.In POV of Line 11 you can see it between Pusan Road and Yuqiao.)

Would you like to explain more about it?

**Comment by thekingofcity** [comment]
As we make no fundamental changes in the underlying params, this could be brought back to the to-do list.
